src/accelerate/utils/dataclasses.py

```diff
@@ -1196,13 +1199,13 @@ def set_mixed_precision(self, mixed_precision):
     def set_deepspeed_weakref(self):
```

Comment:
let's move the discussion on zero.Init / zero3_init_flag flag here - so it's self-contained.

Yes, I'm aware of this - but I don't see it being tested in the new multi-model tests. how do you know that it works correctly?

Reply:
Thanks. 

And true, will do a very quick test rn. Thanks for catching it

Reply:
added tests, and this does indeed work :)